### PR TITLE
Remove docker-compose leftovers

### PR DESCRIPTION
**Pull request by tklauser**
`docker-compose` is no longer used in CI with #14400 merged. Adjust docs, test installation scripts and dev-doctor accordingly. See individual commits for details.

### CODE REVIEW

**Comment by tklauser** [comment]
test-me-please

**Review by twpayne — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]
Thanks!

(Could probably have been a single commit with e.g., `treewide: ...`)

**Comment by tklauser** [comment]
> (Could probably have been a single commit with e.g., `treewide: ...`)

Sure, can squash it before merge.

**Comment by tklauser** [comment]
retest-runtime

**Comment by tklauser** [comment]
retest-gke

**Comment by tklauser** [comment]
retest-gke

**Review by joestringer — APPROVED** [review]

**Review by nathanjsweet — APPROVED** [review]
Thanks for the cleanup!

**Comment by pchaigno** [comment]
The only reason reviews from Janitors were requested is because `tools/` is missing from the codeowners. Fixed in https://github.com/cilium/cilium/pull/14433. Marking this pull request as ready to merge.

**Review by aanm — APPROVED** [review]

**Pull request merged**
